### PR TITLE
Add release notes to docs

### PR DESCRIPTION
**Pull request by Kludex**
Add release notes, analogous to what `starlette` [does](https://www.starlette.io/release-notes/).

Missing:
- [x] Add hyperlink to PRs reference.
- [x] Remove date and commit hash from sentences.
- [x] Add missing PR numbers.

### CODE REVIEW

**Comment by Kludex** [comment]
I've found this script: https://github.com/MagicStack/uvloop/blob/master/.github/release_log.py
Should we use it or do something similar?

File: setup.py [hunk 1]
```diff
@@ -95,4 +95,5 @@ def get_packages(package):
     [console_scripts]
     uvicorn=uvicorn.main:main
     """,
+    project_urls={"Changelog": "https://www.uvicorn.org/release-notes/"},
```
Comment (euri10) [inline setup.py:98]: what about we just add the github link of the Changelog.md to this ?

Comment (Kludex) [reply setup.py:98]: I think we can start from there. But I think we should have it on the docs as well at some point, with the reference to the PRs, it's easy to travel around.

**Comment by Kludex** [comment]
Closing this in favor of #1139.

**Pull request closed without merging**